x/evm/keeper/grpc_query_test.go

```diff
@@ -1517,31 +1518,37 @@ func (suite *KeeperTestSuite) TestQueryBaseFee() {
 			true,
 		},
 	}
+
+	// Save initial configure to restore it between tests
+	denom := types.GetEVMCoinDenom()
+	decimals := types.GetEVMCoinDecimals()
+	chainConfig := types.DefaultChainConfig(suite.network.GetChainID())
+
 	for _, tc := range testCases {
 		tc := tc
 		suite.Run(tc.name, func() {
 			// Set necessary params
 			tc.setParams()
-
 			// Get the expected response
 			expResp := tc.getExpResp()
-
 			// Function under test
 			res, err := suite.network.GetEvmClient().BaseFee(
 				suite.network.GetContext(),
 				&types.QueryBaseFeeRequest{},
 			)
-
 			if tc.expPass {
 				suite.Require().NotNil(res)
 				suite.Require().Equal(expResp, res, tc.name)
 				suite.Require().NoError(err)
 			} else {
 				suite.Require().Error(err)
 			}
-
 			suite.Require().NoError(suite.network.NextBlock())
-			err = evmconfig.NewEVMConfigurator().
+			configurator := types.NewEVMConfigurator()
+			configurator.ResetTestChainConfig()
+			err = configurator.
+				WithChainConfig(chainConfig).
+				WithEVMCoinInfo(denom, decimals).
```

Comment:
@GAtom22 is this the way to use the new configurator? The zero fee test was throwing a panic because the EVMConfig was nil when trying to finalize the next block. So I'm reseting to the original values between each test 

The test that changes the values is reseting again the values inside the setParams function, so this solution doesn't conflict with that test

Reply:
yeah, If you need to set a custom config, you need to do that, Reset and then configure again, the chainConfig and the EvmCoinInfo